rollup/circuitcapacitychecker/impl.go

```diff
@@ -73,15 +73,12 @@ func (ccc *CircuitCapacityChecker) ApplyTransaction(traces *types.BlockTrace) (*
 	}
 
 	tracesStr := C.CString(string(tracesByt))
-	defer func() {
-		C.free(unsafe.Pointer(tracesStr))
-	}()
+	defer C.free(unsafe.Pointer(tracesStr))
```

Comment:
how does this change make it different?

Reply:
There are some [subtle differences](https://stackoverflow.com/questions/47321012/difference-between-defer-call-a-function-directly-and-call-it-in-an-anonymous-fu), I mainly refer to [scroll's update](https://github.com/scroll-tech/scroll/pull/998/files#diff-d46c91e57eb218f78dd21be2e9a5042e789c0855a51160b77023e7efe92a3f10)